tests/unit/test_cubeset.py

```diff
@@ -0,0 +1,176 @@
+        _cubes_prop_.return_value = (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        has_means = cube_set.has_means
+
+        assert has_means == expected_value
+
+    def it_knows_whether_it_has_weighted_counts(
+        self, has_weighted_counts_fixture, _cubes_prop_, cube_
+    ):
+        first_cube_has_w_counts, expected_value = has_weighted_counts_fixture
+        cube_.is_weighted = first_cube_has_w_counts
+        _cubes_prop_.return_value = (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        is_weighted = cube_set.has_weighted_counts
+
+        assert is_weighted == expected_value
+
+    def it_knows_its_name(self, has_name_fixture, _cubes_prop_, cube_):
+        first_cube_name, expected_value = has_name_fixture
+        cube_.name = first_cube_name
+        _cubes_prop_.return_value = (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        name = cube_set.name
+
+        assert name == expected_value
+
+    def it_knows_its_description(self, has_description_fixture, _cubes_prop_, cube_):
+        first_cube_description, expected_value = has_description_fixture
+        cube_.description = first_cube_description
+        _cubes_prop_.return_value = (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        description = cube_set.description
+
+        assert description == expected_value
+
+    def it_can_show_pairwise(self, can_show_pairwise_fixture, _cubes_prop_, cube_):
+        ndim, expected_value = can_show_pairwise_fixture
+        cube_.ndim = ndim
+        _cubes_prop_.return_value = ndim * (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        can_show_pairwise = cube_set.can_show_pairwise
+
+        assert can_show_pairwise == expected_value
+
+    def it_knows_its_missing_count(self, missing_count_fixture, _cubes_prop_, cube_):
+        first_cube_missing_count, expected_value = missing_count_fixture
+        cube_.missing = first_cube_missing_count
+        _cubes_prop_.return_value = (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        missing_count = cube_set.missing_count
+
+        assert missing_count == expected_value
+
+    def it_knows_when_it_is_ca_as_0th(self, is_ca_as_0th_fixture, _cubes_prop_, cube_):
+        ncubes, expected_value = is_ca_as_0th_fixture
+        cubes_ = (cube_,) * ncubes
+        cubes_[0].dimension_types = (DT.CA_SUBVAR,) * ncubes
+        _cubes_prop_.return_value = cubes_
+        cube_set = CubeSet(cubes_, None, None, None)
+
+        isca_as_0th = cube_set.is_ca_as_0th
+
+        assert isca_as_0th == expected_value
+
+    def it_has_the_right_partition_set(
+        self, partition_set_fixture, _cubes_prop_, cube_
+    ):
+        partition_set, expected_value = partition_set_fixture
+        cube_.partitions = partition_set
+        _cubes_prop_.return_value = (cube_,)
+        cube_set = CubeSet(None, None, None, None)
+
+        partitions_set = cube_set.partition_sets
```

Comment:
`partition_sets`, not `partitions_set`. I think the item arriving from the fixture is `cube_partitions`, not `partition_set.`